### DISCUSSION TITLE
How does OpenROAD compute the floorplan size while using core_utilization

### DISCUSSION THREAD

**Discussion by lustefan97** [The-OpenROAD-Project/OpenROAD#4232] [Q&A] [answered]
Hello,

I have a quick question out of pure curiosity : How does OpenROAD compute the die area when using the core_utilization argument in the `initialize_floorplan`?

**Comment by maliberty** [comment, marked as answer]
The code starts at https://github.com/The-OpenROAD-Project/OpenROAD/blob/2bdae1c13b8c6387644f12f760254f472b04f4e7/src/ifp/src/InitFloorplan.cc#L126

In short we take the area of the instances coming from synthesis and convert that to a core area based on the utilization.  We then shape that core area based on the aspect_ratio.  Finally we expand it by the spacing on each edge to get the die area.

**Reply by lustefan97** [reply]
I see, thanks for the explanation !

**Reply by lustefan97** [reply]
@maliberty I have an issue that I think is linked to the floorplanning step, but reported in the global placement step. I use the ASAP7 PDK, I have a dummy design for which I specify a target design utilization during the floorplanning step. 

Essentially I try to increase the design utilization, here I target a DU of 70% thus I have `set core_utilization 70` (and I am trying to go to 80%). The problem is that during the global placement step, this information is reported :

[INFO GPL-0019] Util(%): 115.63

Which lead to this error : 

[ERROR GPL-0301] Utilization exceeds 100%.

I don't understand how, based off of how the floorplan size is computed, this utilization value is obtained since it should also be based off of the sum of the cells area, and for below 70 (tried with target DU of 60 and 65) it was fine...

**Reply by maliberty** [reply]
My guess is it relates to padding.  Can you try setting
```
export CELL_PAD_IN_SITES_GLOBAL_PLACEMENT = 0
export CELL_PAD_IN_SITES_DETAIL_PLACEMENT = 0
```
and see if that explains the issue?

**Reply by lustefan97** [reply]
It did indeed explains the issue, GPL do now have a reported DU that is similar to the target one. May I ask why did padding the design left and right cause such an issue ? I thought that padding would add to the originally computed die size, not "reduce" the available space for placement. The padding argument are not explained in the GPL documentation either...

**Reply by rovinski** [reply]
Padding makes cells appear larger to GPL (and DPL) in order to prevent placing cells too close to each other. These larger cell sizes will cause it to look like the utilization is higher because there is less free space.
